client/src/node.rs

```diff
@@ -123,6 +136,49 @@ impl Node {
         Ok(())
     }
 
+    async fn get_payloads(&self, start_slot: u64, end_slot: u64) -> Result<Vec<ExecutionPayload>, NodeError> {
+        let payloads_fut = (start_slot..end_slot).rev().into_iter().map(|slot| async move {
+            let self_ref = self;
+            let inner_closure = async move {
+                let payload = self_ref
+                    .consensus
+                    .get_block_from_rpc(slot)
+                    .await
+                    .map_err(NodeError::ConsensusPayloadError)?
+                    .body
+                    .execution_payload;
+                Ok::<ExecutionPayload, NodeError>(payload)
+            };
+            inner_closure.await
+        });
+        let mut prev_parent_hash: Option<Bytes32> = None;
+        let mut payloads: Vec<ExecutionPayload> = Vec::new();
+        for result in join_all(payloads_fut).await {
+            if result.is_err() {
+                // If fetching any payload failed we only backfill up to that payload
+                // TODO: Review if it is better to error out of the whole payload update instead;
+                break;
+            }
+            let payload = result?;
+            if let Some(parent_block_hash) = &prev_parent_hash {
+                if &payload.block_hash != parent_block_hash {
```

Comment:
I think this will break if a slot is missed at any point in the chain. I'm not exactly sure what `rpc.get_block` returns in the case of a missed slot, but we should find out and handle that case.

Reply:
Good point. I don't know what the behaviour would be in that case either. Will try to find a skipped block and find out

Reply:
Another way we could do it would be just sequentially requesting the next block using the current `parent_block_hash`, which would probably fix this issue, but in that case we couldn't do the rpc requests concurrently.

Reply:
Looks like the consensus rpc returns a 404 Error when requesting a skipped block:
<img width="612" alt="image" src="https://user-images.githubusercontent.com/15629702/216758588-83deab90-5c17-416a-99a6-359488b0eec4.png">

Reply:
This means that in the current implementation it will stop backfilling at that point due to the break statement [here](          https://github.com/a16z/helios/blob/0822f5874a759d6673be31c74d43417adf25acb6/consensus/src/consensus.rs#L123)

Reply:
I think this is the source of some intermittent errors that I had observed and which I assumed to be due to rate limiting.

Reply:
The easiest way to (hopefully) fix this is to just continue with the next block in case any rpc request failed. 
This should allow it to continue working if there is a skipped block, although the effectively backfilled history would be 1 element shorter than `self.history_size` in that case, which should not be too much of an issue I guess.

See: 
https://github.com/a16z/helios/pull/189/commits/48499addbb84bff58b9461376554ead47fcf35d3